php/commands/cli.php

```diff
@@ -1,575 +1,3 @@
-				}
-				$spec[$key]['current'] = $current;
-			}
-		}
-
-		if ( 'var_export' === \WP_CLI\Utils\get_flag_value( $assoc_args, 'format' ) ) {
-			var_export( $spec );
-		} else {
-			echo json_encode( $spec );
-		}
-	}
-
-	/**
-	 * Dump the list of installed commands, as JSON.
-	 *
-	 * ## EXAMPLES
-	 *
-	 *     # Dump the list of installed commands.
-	 *     $ wp cli cmd-dump
-	 *     {"name":"wp","description":"Manage WordPress through the command-line.","longdesc":"\n\n## GLOBAL PARAMETERS\n\n  --path=<path>\n      Path to the WordPress files.\n\n  --ssh=<ssh>\n      Perform operation against a remote server over SSH.\n\n  --url=<url>\n      Pretend request came from given URL. In multisite, this argument is how the target site is specified. \n\n  --user=<id|login|email>\n
-	 *
-	 * @subcommand cmd-dump
-	 */
-	public function cmd_dump() {
-		echo json_encode( self::command_to_array( WP_CLI::get_root_command() ) );
-	}
-
-	/**
-	 * Generate tab completion strings.
-	 *
-	 * ## OPTIONS
-	 *
-	 * --line=<line>
-	 * : The current command line to be executed.
-	 *
-	 * --point=<point>
-	 * : The index to the current cursor position relative to the beginning of the command.
-	 *
-	 * ## EXAMPLES
-	 *
-	 *     # Generate tab completion strings.
-	 *     $ wp cli completions --line='wp eva' --point=100
-	 *     eval
-	 *     eval-file
-	 */
-	public function completions( $_, $assoc_args ) {
-		$line = substr( $assoc_args['line'], 0, $assoc_args['point'] );
-		$compl = new \WP_CLI\Completions( $line );
-		$compl->render();
-	}
-
-	/**
-	 * List available WP-CLI aliases.
-	 *
-	 * Aliases are shorthand references to WordPress installs. For instance,
-	 * `@dev` could refer to a development install and `@prod` could refer to
-	 * a production install. This command gives you visibility in what
-	 * registered aliases you have available.
-	 *
-	 * ## OPTIONS
-	 *
-	 * [--format=<format>]
-	 * : Render output in a particular format.
-	 * ---
-	 * default: yaml
-	 * options:
-	 *   - yaml
-	 *   - json
-	 * ---
-	 *
-	 * ## EXAMPLES
-	 *
-	 *     # List all available aliases.
-	 *     $ wp cli alias
-	 *     ---
-	 *     @all: Run command against every registered alias.
-	 *     @prod:
-	 *       ssh: runcommand@runcommand.io~/webapps/production
-	 *     @dev:
-	 *       ssh: vagrant@192.168.50.10/srv/www/runcommand.dev
-	 *     @both:
-	 *       - @prod
-	 *       - @dev
-	 *
-	 * @alias aliases
-	 */
-	public function alias( $_, $assoc_args ) {
-		WP_CLI::print_value( WP_CLI::get_runner()->aliases, $assoc_args );
-	}
-
-	/**
-	 * Get a string representing the type of update being checked for.
-	 */
-	private function get_update_type_str( $assoc_args ) {
-		$update_type = ' ';
-		foreach( array( 'major', 'minor', 'patch' ) as $type ) {
-			if ( true === \WP_CLI\Utils\get_flag_value( $assoc_args, $type ) ) {
-				$update_type = ' ' . $type . ' ';
-				break;
-			}
-		}
-		return $update_type;
-	}
-
-}
-
 WP_CLI::add_command( 'cli', 'CLI_Command' );
```

Comment:
I was looking for a command that would not be split out, to use for the tests. So, the tests do test all of the different ways to override this command.

For this to work, however, the command itself needs to be autoloaded on-demand, not immediately loaded upon bootstrap.